fdbserver/worker.actor.cpp

```diff
@@ -2017,6 +2030,57 @@ ACTOR Future<Void> workerServer(Reference<IClusterConnectionRecord> connRecord,
 				logData.back().actor = oldLog.getFuture() || tl;
 				logData.back().uid = s.storeID;
 				errorForwarders.add(forwardError(errors, Role::SHARED_TRANSACTION_LOG, s.storeID, tl));
+			} else if (s.storedComponent == DiskStore::BlobWorker) {
+				if (blobWorkerFuture.isReady()) {
+					LocalLineage _;
+					getCurrentLineage()->modify(&RoleLineage::role) = ProcessClass::ClusterRole::BlobWorker;
+
+					BlobWorkerInterface recruited(locality, deterministicRandom()->randomUniqueID());
+					recruited.initEndpoints();
+
+					std::map<std::string, std::string> details;
+					details["StorageEngine"] = s.storeType.toString();
+					startRole(Role::BLOB_WORKER, recruited.id(), interf.id(), details, "Restored");
+
+					DUMPTOKEN(recruited.waitFailure);
+					DUMPTOKEN(recruited.blobGranuleFileRequest);
+					DUMPTOKEN(recruited.assignBlobRangeRequest);
+					DUMPTOKEN(recruited.revokeBlobRangeRequest);
+					DUMPTOKEN(recruited.granuleAssignmentsRequest);
+					DUMPTOKEN(recruited.granuleStatusStreamRequest);
+					DUMPTOKEN(recruited.haltBlobWorker);
+					DUMPTOKEN(recruited.minBlobVersionRequest);
```

Comment:
Nit: not a blocker for this PR, we can do later, but this is missing flushGranuleRequest